src/core/dom/image/interface.ts

```diff
@@ -0,0 +1,334 @@
+/*!
+ * V4Fire Client Core
+ * https://github.com/V4Fire/Client
+ *
+ * Released under the MIT license
+ * https://github.com/V4Fire/Client/blob/master/LICENSE
+ */
+
+import iBlock from 'super/i-block/i-block';
+import { VNodeDirective } from 'core/component/engines';
+
+export interface DirectiveOptions extends VNodeDirective {
+	modifiers: {
+		[key: string]: boolean;
+	};
+
+	value?: DirectiveValue;
+}
+
+export interface ImageOptions {
+	/**
+	 * Execution context
+	 *
+	 * An async module will be used from the passed context to set events.
```

Comment:
Execution context. The context is used to provide a component environment, like, `async`, event emitters, etc. When API is used as a directive,  the context will automatically taken from a VNode instance.